### PR TITLE
Extending basic build support 

### PR DESCRIPTION
**Pull request by smurawski**
This PR adds basic build support (context and dockerfile) via Azure Container Registry (ACR) build tasks. If a service in the Compose file has the build key defined, either the provided ACR (or a new ACR which will be automatically provisioned) will be used to build the source container image.  The container registry will be mapped the container app to finish the deployment scenario.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az containerapp compose

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by smurawski** [comment]
@zhoxing-ms this pr is ready for review. Please let me know if there is anything else I need to do to release a new version.

File: src/containerapp-compose/HISTORY.rst [hunk 1]
```diff
@@ -5,4 +5,9 @@ Release History
...
 0.1.0
 ++++++
-* Initial release.
\ No newline at end of file
+* Initial release.
+
+0.2.0
+++++++
+* Add basic build support
+* Fix bug in specifying memory
```
Comment (zhoxing-ms) [inline src/containerapp-compose/HISTORY.rst]: Please write the description of the new version at the top of the history note

Comment (smurawski) [reply src/containerapp-compose/HISTORY.rst]: I'm not exactly sure what you mean, but I tried to clarify the first item which is the new feature description.

Comment (zhoxing-ms) [reply src/containerapp-compose/HISTORY.rst]: Please refer to [HISTORY.md](https://github.com/Azure/azure-cli-extensions/blob/main/src/spring/HISTORY.md) to write the description of the new version above the description of the old version

Comment (smurawski) [reply src/containerapp-compose/HISTORY.rst]: Now I see @zhoxing-ms .  Thank you, I misunderstood the direction.  I made the adjustment.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**